backend/src/main/routes/AccountRoute.ts

```diff
@@ -40,51 +42,60 @@ export default class AccountRoute extends CommonRoutesConfig {
+          try {
+            const employeeAccounts = await this.employeeAccountService.getAllEmployeeAccounts();
+            if (employeeAccounts === null || employeeAccounts.length === 0) {
+              next(new HttpException(StatusCodes.NOT_FOUND, ReasonPhrases.NOT_FOUND));
+            } else {
+              const employeeAccountsDTOs: Array<any> = [];
+              employeeAccounts?.forEach((employeeAccount) => {
+                const dto = JSON.parse(JSON.stringify(employeeAccount));
+                delete dto.account.password;
+                employeeAccountsDTOs.push(dto);
+              });
+              res.status(StatusCodes.OK).send(employeeAccountsDTOs);
+            }
+          } catch (err) {
+            next(err);
           }
-        } catch (err) {
-          next(err);
         }
-      });
+      );
 
     this.getApp()
       .route(`/accounts/allEmployees/:business`)
-      .get(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-        try {
-          const employeeAccounts = await this.employeeAccountService.getAllEmployeeAccountsByBusiness(
-            req.params.business
-          );
+      .get(
+        checkJwt,
+        checkRole(new Set([Roles.BUSINESS])),
+        async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+          try {
+            const employeeAccounts = await this.employeeAccountService.getAllEmployeeAccountsByBusiness(
+              req.params.business
+            );
 
-          if (employeeAccounts === null || employeeAccounts.length === 0) {
-            next(new HttpException(StatusCodes.NOT_FOUND, ReasonPhrases.NOT_FOUND));
-          } else {
-            const employeeAccountsDTOs: Array<any> = [];
-            employeeAccounts?.forEach((employeeAccount) => {
-              const dto = JSON.parse(JSON.stringify(employeeAccount));
-              delete dto.account.password;
-              employeeAccountsDTOs.push(dto);
-            });
-            res.status(StatusCodes.OK).send(employeeAccountsDTOs);
+            if (employeeAccounts === null || employeeAccounts.length === 0) {
+              next(new HttpException(StatusCodes.NOT_FOUND, ReasonPhrases.NOT_FOUND));
+            } else {
+              const employeeAccountsDTOs: Array<any> = [];
+              employeeAccounts?.forEach((employeeAccount) => {
+                const dto = JSON.parse(JSON.stringify(employeeAccount));
+                delete dto.account.password;
+                employeeAccountsDTOs.push(dto);
+              });
+              res.status(StatusCodes.OK).send(employeeAccountsDTOs);
+            }
+          } catch (err) {
+            next(err);
           }
-        } catch (err) {
-          next(err);
         }
-      });
+      );
 
     this.getApp()
       .route(`/accounts/employee/:email`)
+      .all(checkJwt, checkRole(new Set([Roles.BUSINESS])))
```

Comment:
similar issue to the one mentioned below.

Reply:
Fixed. Added restriction for employees.